lib/iris/pandas.py

```diff
@@ -116,6 +196,291 @@ def as_cube(pandas_array, copy=True, calendars=None):
+    >>> converted_cube = as_cubes(my_series)[0]
+    >>> print(converted_cube.data)
+    [300.  nan 302.]
+    >>> converted_cube.data = np.ma.masked_invalid(converted_cube.data)
+    >>> print(converted_cube.data)
+    [300.0 -- 302.0]
+
+    If the :class:`~pandas.DataFrame` uses columns as as a second dimension,
+    :func:`pandas.melt` should be used to convert the data to the expected
+    n-dimensional format :
+
+    >>> my_df = DataFrame({
+    ...     "latitude": [35, 25],
+    ...     0: [300, 301],
+    ...     10: [302, 303],
+    ... })
+    >>> print(my_df)
+       latitude    0   10
+    0        35  300  302
+    1        25  301  303
+    >>> my_df = my_df.melt(
+    ...     id_vars=["latitude"],
+    ...     value_vars=[0, 10],
+    ...     var_name="longitude",
+    ...     value_name="air_temperature"
+    ... )
+    >>> print(my_df)
+       latitude longitude  air_temperature
+    0        35         0              300
+    1        25         0              301
+    2        35        10              302
+    3        25        10              303
+    >>> my_df = my_df.set_index(["latitude", "longitude"])
+    >>> my_df = my_df.sort_index()
+    >>> converted_cube = as_cubes(my_df)[0]
+    >>> print(converted_cube)
+    air_temperature / (unknown)         (latitude: 2; longitude: 2)
+        Dimension coordinates:
+            latitude                             x             -
+            longitude                            -             x
+
+    """
+    calendars = calendars or {}
+    aux_coord_cols = aux_coord_cols or []
+    cell_measure_cols = cell_measure_cols or []
+    ancillary_variable_cols = ancillary_variable_cols or []
+
+    is_series = isinstance(pandas_structure, pandas.Series)
+
+    if copy:
+        pandas_structure = pandas_structure.copy()
+
+    pandas_index = pandas_structure.index
+    if not pandas_index.is_unique:
+        message = (
+            f"DataFrame index ({pandas_index.names}) is not unique per "
+            "row; cannot be used for DimCoords."
+        )
+        raise ValueError(message)
+
+    if not pandas_index.is_monotonic:
+        # Need monotonic index for use in DimCoord(s).
+        # This function doesn't sort_index itself since that breaks the
+        #  option to return a data view instead of a copy.
+        message = (
+            "Pandas index is not monotonic. Consider using the "
+            "sort_index() method before passing in."
+        )
+        raise ValueError(message)
+
+    cube_shape = getattr(pandas_index, "levshape", (pandas_index.nunique(),))
+
+    cube_kwargs = {}
+
+    def format_dimensional_metadata(dm_class_, values_, name_, dimensions_):
+        # Common convenience to get the right DM in the right format for
+        #  Cube creation.
+        calendar = calendars.get(name_)
+        instance = _get_dimensional_metadata(
+            name_, values_, calendar, dm_class_
+        )
+        return (instance, dimensions_)
```

Comment:
It doesn't look like you use `dimensions_` in this function, you just pass it out again, so you could remove it from here

Reply:
Part of the common convenience of `format_dimensional_metadata()` is getting the objects out in the correct format, since I do this more than once in the main code. That includes placing the object and its dimensions into a `tuple`.

Is that a bit strange? Should I just be creating the appropriate `tuple` each time, outside this function?

Reply:
I see, that makes sense. I think it's fine to keep it as it is. As you say you are getting it in the right format. I had actually briefly wondered how best to write that if it weren't included in this function. I think it makes sense to do it here